### ISSUE TITLE
Deprecate in favor of nix

### ISSUE THREAD

**Issue by Susurrus** [dcuddeback/termios-rs#12]
The new `nix` 0.9 release includes a new termios API that covers the same functionality provided by this crate. Depending on your workload, it might be worth deprecating this crate and stating that people should use `nix` directly instead.

**Comment by conradkdotcom** [comment]
@Susurrus I think we should keep `termios` for the following reasons.

tl;dr: Both `nix` and `termios` have their use. `termios` is easy to audit. Both should be kept around.

- including spaces/blanks/comments, `termios` has 5K LOCs, `nix` has 39K LOCs
- `termios` depends only on `libc`, `nix` has a hard dependency on `bitflags`, `cfg-if`, `libc` and `void` and dev dependency on `lazy_static`, ` nix-test`, `rand`, `tempdir`, `tempfile`
- `termios` requires little maintenance, so little updates in `Cargo.lock`

Overall, `termios` is much easier to audit. I use it in a password manager and so I'm very wary of having large dependencies. I like the relative simplicity of `termios` 😃

**Comment by conradkdotcom** [comment]
I don't think the 2 crates can be compared. `nix` is a huge beast, powerful but also difficult to audit. `termios-rs` on the other hand is less powerful but easy to audit because of its small size.

Seems like both use MIT licensing though, so that would make deprecation easy.

**Comment by conradkdotcom** [comment]
I've just noticed that `libc` now includes all of `termios-rs` and it's unit testing for a lot of different OSes. For instance, for `TCSANOW`, see: https://github.com/rust-lang/libc/search?utf8=%E2%9C%93&q=TCSANOW&type=

**Comment by Susurrus** [comment]
@conradkdotcom Yes, I finished adding support for the `termios` API to `nix` (which included pushing all FFI declarations into `libc`) because I wanted to have solid cross-platform support for it. The CI story for both `libc` and `nix` are both very good. Additionally `nix` aims to support all POSIX platforms also supported by `libc` (see [the README](https://github.com/nix-rust/nix) for platform support promises).

>I don't think the 2 crates can be compared. nix is a huge beast, powerful but also difficult to audit. termios-rs on the other hand is less powerful but easy to audit because of its small size.

`nix` is a bunch of disparate APIs that share very little code, so I don't think yours is an appropriate characterization. If you look at [the code for the termios API](https://github.com/nix-rust/nix/blob/master/src/sys/termios.rs) you'll actually see it only builds on 3 datatypes defined by `nix`: `Pid`, `Result`, and `Error`. Auditing this codebase should be no more onerous than that of `termios-rs` given that `tokei` reports the LoC for both projects' APIs to be very similar.

**Comment by dcuddeback** [comment]
I don't see any reason to deprecate `termios-rs`. It requires very little maintenance and has fewer dependencies than `nix`, which some people might prefer.

Some developers (such as myself) consider the cost (in terms of complexity and maintenance) before adding a dependency to a project. For those developers, less is more, and `termios-rs` offers a simple API that's focused on a single piece of POSIX functionality and has only `libc` as a dependency. Some developers see dependencies as a near-zero cost to their project. For them (or in some circumstances such as prototyping), the extra surface area and transitive dependencies of `nix` might not be a big deal.

It's nice to have choices.

@conradkdotcom Thanks for pointing out that `libc` has the termios definitions now. I may actually switch `serial-rs` to use that instead to reduce dependencies.
